### PR TITLE
fix(curriculum): cash register test 17

### LINKED ISSUE #56241: Cash register: Test 17 fails sometimes
### Describe the Issue

Test 17 is incorrect sometimes, it fails around 10% of the time.

One incorrect case is
```
cash = 50
price = 22.7
cid = [
  [ 'PENNY', 0 ],
  [ 'NICKEL', 0 ],
  [ 'DIME', 0.7 ],
  [ 'QUARTER', 2 ],
  [ 'ONE', 1 ],
  [ 'FIVE', 55 ],
  [ 'TEN', 70 ],
  [ 'TWENTY', 140 ],
  [ 'ONE HUNDRED', 0 ]
]
```
result is `OPEN TWENTY: $20.00 FIVE: $5.00 ONE: $1.00 QUARTER: $1.00 DIME: $0.30`
which is correct based on the input, but it's supposed to be `INSUFFICIENT_FUNDS` according to the test

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/build-a-cash-register-project/build-a-cash-register

### Your code

```js

let price = 79.35;
let cid = [ [ 'PENNY', 0.03 ],
  [ 'NICKEL', 0 ],
  [ 'DIME', 0.9 ],
  [ 'QUARTER', 3.75 ],
  [ 'ONE', 7 ],
  [ 'FIVE', 70 ],
  [ 'TEN', 90 ],
  [ 'TWENTY', 200 ],
  [ 'ONE HUNDRED', 1400 ] ];

const CurrencyValue = [
  1,
  5,
  10,
  25,
  100,
  500,
  1000,
  2000,
  10000,
];

const cashInput = document.getElementById("cash");
const purchaseButton = document.getElementById("purchase-btn");
const changeDueSpan = document.getElementById("change-due");

function calcChangeDue(cash) {
  function currenciesToString(arr) {
    return arr.reduce((str, val) => str + (val[1] !== 0 ? ` ${val[0]}: $${(val[1] / 100).toFixed(2)}` : ""), "")
  }

  console.log(cid)

  // Uses integers for arithmetic to avoid floating-point errors
  const cash_pennies = Math.round(cash * 100);

  const cid_pennies = cid.map(elem => [elem[0], Math.round(elem[1] * 100)]);
  const totalDrawer = [cid_pennies[0][1]];

  for (let i = 1; i < cid_pennies.length; i++) {
    totalDrawer.push(totalDrawer[i - 1] + cid_pennies[i][1]);
  }

  // Close if drawer contains the exact amount of change requested
  if (totalDrawer[cid_pennies.length - 1] === cash_pennies) {
    return "CLOSED" + currenciesToString(cid_pennies);
  }

  const recur = (idx, cashLeft) => {
    if (totalDrawer[idx] < cashLeft) {
      return null;
    }

    for (let currency = cid_pennies[idx][1]; currency >= 0; currency -= CurrencyValue[idx]) {
      if (currency > cashLeft) continue;

      if (currency == cashLeft) {
        return [[cid_pennies[idx][0], currency]]
      }
      if (idx > 0) {
        if (currency + totalDrawer[idx - 1] < cashLeft) {
          break;
        }

        const recursive = recur(idx - 1, (cashLeft - currency));
        if (recursive != null) {
          return [[cid_pennies[idx][0], currency], ...recursive];
        }
      }
    }

    return null;
  }
  const result = recur(cid.length-1, cash_pennies);

  if (result == null) {
    return "INSUFFICIENT_FUNDS";
  }

  return "OPEN" + currenciesToString(result);
}

purchaseButton.addEventListener("click", () => {
  const cash = Number.parseFloat(cashInput.value);
  console.log(cash, price);
  if (cash < price) {
    alert("Customer does not have enough money to purchase the item");
  } else if (cash === price) {
    changeDueSpan.innerText = "No change due - customer paid with exact cash";
  } else {
    const changeDue = calcChangeDue(cash - price);

    console.log(changeDue);
    
    changeDueSpan.innerText = "Status: " + changeDue;
  }
  console.log();
});

```

### Expected behavior

The individual denomination amounts should make it impossible to return the needed change.

### System

- Device: Desktop
- OS: Windows 10
- Browser: Firefox

### PR DESCRIPTION
**Pull request by bulutyerli**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56241

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md [hunk 1]
```diff
@@ -461,15 +461,23 @@ cashInput.value = `${randomCash / 100}`;
...
-    changeLeft % denomination === 0 ? Math.min(15, maxCountInChange - 1) : 15
-  );
+
+  let drawerCount;
+  if (i === 0) {
+    drawerCount = _randomNumber(Math.min(15, maxCountInChange));
+  } else {
+    const previousDrawerTotal = _cashInDrawer[_cashInDrawer.length - 1][1] * 100;
+    const maxDrawerCount = Math.floor(previousDrawerTotal / denomination);
+    drawerCount = _randomNumber(Math.min(15, Math.min(maxCountInChange, maxDrawerCount - 1)));
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:475]: If I'm reading it correctly, `drawerCount` will always be `0` after amount of the previous denomination was `0`.

Comment (bulutyerli) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:475]: Yes you are right, i need to think more about this i guess, for now i close the pr

**Review by Sembauke — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Comment by ilenia-magoni** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**